### PR TITLE
JNG-4565 Create two static query tests (JNG-4563, JNG-4564)

### PR DESCRIPTION
**Pull request by fmagnucz**
<table>
<td>
  <a href="https://blackbelt.atlassian.net/browse/JNG-4565" title="JNG-4565" target="_blank">StoryJNG-4565</a>  Implement two query tests
  </td></table>
  <br />

### CODE REVIEW

File: model-test/src/test/java/hu/blackbelt/judo/meta/jsl/runtime/StaticQueryTests.xtend [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+            query MyEntity[] queryByMyEnum(MyEnum p1 = MyEnum#A00) =>
+                MyEntity!all()
+                !filter(e | e.fEnum == p1);
+        '''.parse => [
+            assertNoErrors
+        ]
+    }
+    
+    // TODO: JNG-4566
+    // @Test
```
Comment (bencegelei) [inline model-test/src/test/java/hu/blackbelt/judo/meta/jsl/runtime/StaticQueryTests.xtend]: >  `|`
> `V`
```
// TODO: JNG-4566
// @Test
```

Please use the following format for disabled tests:
```
@Test
@Disabled("JNG-4566")
```

**Review by bencegelei — CHANGES_REQUESTED** [review]

File: model-test/src/test/java/hu/blackbelt/judo/meta/jsl/runtime/StaticQueryTests.xtend (hunk 1 above)
Comment (fmagnucz) [reply model-test/src/test/java/hu/blackbelt/judo/meta/jsl/runtime/StaticQueryTests.xtend]: OK

**Review by bencegelei — APPROVED** [review]

File: model-test/src/test/java/hu/blackbelt/judo/meta/jsl/runtime/StaticQueryTests.xtend [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+        "REQ-ENT-011",
+        "REQ-EXPR-001",
+        "REQ-EXPR-002",
+        "REQ-EXPR-004",
+        "REQ-EXPR-006",
+        "REQ-EXPR-007",
+        "REQ-EXPR-008",
+        "REQ-EXPR-022"
+    ])
+    def void tcs018() {
```
Comment (noherczeg) [inline model-test/src/test/java/hu/blackbelt/judo/meta/jsl/runtime/StaticQueryTests.xtend]: This is a cryptic method name. We should use proper descriptive names. We also don't know what it tests. Having 26 requirements tested in a single test case looks a bit unrealistic to me also.

File: model-test/src/test/java/hu/blackbelt/judo/meta/jsl/runtime/StaticQueryTests.xtend [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+            type date Date;
+            type time Time;
+            type timestamp Timestamp;
+            type numeric Integer(precision = 9, scale = 0);
+            type numeric Long(precision = 15, scale = 0);
+            type string String(min-size = 0, max-size = 4000);
+            
+            // my enum
+            enum MyEnum {
+                A01 = 1;
```
Comment (noherczeg) [inline model-test/src/test/java/hu/blackbelt/judo/meta/jsl/runtime/StaticQueryTests.xtend]: Cryptic enum literals.

File: model-test/src/test/java/hu/blackbelt/judo/meta/jsl/runtime/StaticQueryTests.xtend [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+        "REQ-EXPR-002",
+        "REQ-EXPR-004",
+        "REQ-EXPR-006",
+        "REQ-EXPR-007",
+        "REQ-EXPR-008",
+        "REQ-EXPR-013",
+        "REQ-EXPR-014",
+        "REQ-EXPR-022"
+    ])
+    def void tcs019() {
```
Comment (noherczeg) [inline model-test/src/test/java/hu/blackbelt/judo/meta/jsl/runtime/StaticQueryTests.xtend]: Same here.

**Review by noherczeg — CHANGES_REQUESTED** [review]

File: model-test/src/test/java/hu/blackbelt/judo/meta/jsl/runtime/StaticQueryTests.xtend (hunk 2 above)
Comment (fmagnucz) [reply model-test/src/test/java/hu/blackbelt/judo/meta/jsl/runtime/StaticQueryTests.xtend]: I know, and I agree with you. This moment here are the test descriptions: https://github.com/BlackBeltTechnology/judo-community/tree/requirement_analysis/judo-testcase-specifications

This test case: https://github.com/BlackBeltTechnology/judo-community/blob/requirement_analysis/judo-testcase-specifications/tcs-018/tcs-018.adoc
I think about it, how write and where these information. I am create a demo, and I show it for the team.

File: model-test/src/test/java/hu/blackbelt/judo/meta/jsl/runtime/StaticQueryTests.xtend (hunk 3 above)
Comment (fmagnucz) [reply model-test/src/test/java/hu/blackbelt/judo/meta/jsl/runtime/StaticQueryTests.xtend]: It has not any meaning. It is used only for the test.

File: model-test/src/test/java/hu/blackbelt/judo/meta/jsl/runtime/StaticQueryTests.xtend (hunk 4 above)
Comment (fmagnucz) [reply model-test/src/test/java/hu/blackbelt/judo/meta/jsl/runtime/StaticQueryTests.xtend]: https://github.com/BlackBeltTechnology/judo-community/blob/requirement_analysis/judo-testcase-specifications/tcs-019/tcs-019.adoc

File: model-test/src/test/java/hu/blackbelt/judo/meta/jsl/runtime/StaticQueryTests.xtend (hunk 2 above)
Comment (noherczeg) [reply model-test/src/test/java/hu/blackbelt/judo/meta/jsl/runtime/StaticQueryTests.xtend]: Test method names should always describe what they are testing. We don't have written rules for this, but it was a pre-established convention which we should follow.

File: model-test/src/test/java/hu/blackbelt/judo/test/specification/TCS018.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
+package hu.blackbelt.judo.test.specification;
+
+public interface TCS018 {
```
Comment (bencegelei) [inline model-test/src/test/java/hu/blackbelt/judo/test/specification/TCS018.java]: I don't understand why we need a separate interface and implementation. Could there be another implementations of the same test? If not then the "specification" below can be moved to the actual implementation.

If the idea was to separate the specification and the test implementation to decrease lines of code in a file,  I don't think the use of interfaces is proper solution.

File: model-test/src/test/java/hu/blackbelt/judo/test/specification/TCS019.java [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+     * query MyEntity[] queryByDecimal(
+     *     Decimal p1 = (-1526.225)!round(scale = 2)
+     * ) => MyEntity!all()
+     *      !filter(e | e.fDecimal == p1);
+     * 
+     * query MyEntity[] queryByMyEnum(
+     *     MyEnum p1 = (1 <= 2 ? MyEnum#A00 : MyEnum#A03)
+     * ) => MyEntity!all()
+     *      !filter(e | e.fEnum == p1);
+	 *  </pre>
```
Comment (bencegelei) [inline model-test/src/test/java/hu/blackbelt/judo/test/specification/TCS019.java]: In this case almost the actual test implementation is written in a documentation of an interface's method.

**Review by bencegelei — CHANGES_REQUESTED** [review]

**Review by robertcsakany — APPROVED** [review]

**Review by noherczeg — APPROVED** [review]

**Pull request merged**